service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -243,4 +257,24 @@ private String buildPriorPrescription(MedicationRequest medicationRequest) {
         }
         return StringUtils.EMPTY;
     }
+
+    private String buildParticipant(MedicationRequest medicationRequest) {
+        var isPractitioner = buildPredicateReferenceIsA(ResourceType.Practitioner);
+        var isPractitionerRole = buildPredicateReferenceIsA(ResourceType.PractitionerRole);
+        var isOrganization = buildPredicateReferenceIsA(ResourceType.Organization);
+        Predicate<Reference> isRelevant = isPractitioner.or(isPractitionerRole).or(isOrganization);
+
+        if (medicationRequest.hasRecorder() && medicationRequest.getRecorder().hasReference()) {
+            final var reference = medicationRequest.getRecorder();
+            if (reference.hasReferenceElement() && isRelevant.test(reference)) {
+                return participantMapper.mapToParticipant(
+                    messageContext.getIdMapper().get(reference), ParticipantType.AUTHOR);
+            }
+        }
+        return StringUtils.EMPTY;
```

Comment:
I think we're missing a test covering a `medicationRequest` without a reference, and returning `StringUtils.EMPTY`

Reply:
A recorder without a reference would be an empty object in the json. Do we still want to test for it? There is a precedent; search regex `" ?: ?\{\s*}` gives 12 matches, mostly from document reference tests

Reply:
Just to add, from the GP connect API (https://gpc-structured-1-5-0.netlify.app/accessrecord_structured_development_medicationrequest.html)
recorder is a mandatory field, thus should we throw an error if recorder is not populated / does not have a reference?